### PR TITLE
remove workshop, cleanup

### PR DESCRIPTION
**Pull request by akotlar**
Assigning to Daniel 2 because the scorecard beacon is tired.

This removes the workshop login option (previously agreed upon with Cotton), which makes the login.html page totally useless; so I've converted the login link to hit the old /login POST endpoint, and converted the POST to a GET. I think this is semantically fine, because no credentials (or other data) is actually sent to that endpoint (as workshop password is kaput), making that endpoint solely issue a redirect.

Since login.html is gone, I also no longer redirect to it. Instead, unauthorized users are redirected to /error, and I refactored this redirect into a function since it's now used identically in 2 places.

I've also imported the jwt library, so that jwt.exceptions.InvalidTokenError is in scope, and made some minor cleanup.

cc @cseed

### CODE REVIEW

File: notebook2/notebook/notebook.py [hunk 1]
```diff
@@ -16,12 +16,11 @@
 
 import logging
 import os
-import re
```
Comment (akotlar) [inline notebook2/notebook/notebook.py]: not used

File: notebook2/notebook/notebook.py [hunk 2]
```diff
@@ -16,12 +16,11 @@
...
 import requests
 import uuid
 import hashlib
 import kubernetes as kube
-import os
```
Comment (akotlar) [inline notebook2/notebook/notebook.py]: not used

File: notebook2/notebook/notebook.py [hunk 3]
```diff
@@ -16,12 +16,11 @@
...
+import jwt
```
Comment (akotlar) [inline notebook2/notebook/notebook.py]: Python does not import import's modules (submodules), just like Java.

Comment (akotlar) [reply notebook2/notebook/notebook.py]:
```python
from hailjwt import JWTClient, get_domain
# import jwt
raise jwt.exceptions.InvalidTokenError("BLAH")
```
Results in `NameError: name 'jwt' is not defined`

File: notebook2/notebook/notebook.py [hunk 4]
```diff
@@ -143,7 +148,7 @@ def decorated(*args, **kwargs):
             if g.user is None:
                 if for_page:
                     session['referrer'] = request.url
-                    return redirect(external_url_for('login'))
+                    return unauthorized_redirect()
```
Comment (daniel-goldstein) [inline notebook2/notebook/notebook.py]: Why not redirect them to the auth0 login endpoint?

Comment (akotlar) [reply notebook2/notebook/notebook.py]: The reason is that I want to give an indication of why they're there, else it's unexpected behavior, and may annoy. I can't give that message if I send them directly to auth0 currently (there's a way to pass a message to auth0, but that's a bit more involved).

Comment (akotlar) [reply notebook2/notebook/notebook.py]: I could redirect them to a non-generic error page, with a login button. Does that sound better?

**Comment by akotlar** [comment]
@daniel-goldstein I noticed a bug in table.py, patched that in.

File: notebook2/notebook/notebook.py (hunk 4 above)
Comment (akotlar) [reply notebook2/notebook/notebook.py]: Actually... why don't we just try it your way. I don't want 2 clicks to get to notebook.

Comment (akotlar) [reply notebook2/notebook/notebook.py]: One min

Comment (daniel-goldstein) [reply notebook2/notebook/notebook.py]: Makes sense. There's already a login button on the page so I think that's ok. Might be nice to distinguish on the page whether their account is actually unauthorized or if they just need to log back in. Whatever way is easiest to convey that.

**Comment by akotlar** [comment]
The failure is in apiserver somehow, probably trivial. Can't id the problem at the moment because ci is giving 401.

**Pull request merged**